PokeAlarm/Alarms/Telegram/TelegramAlarm.py

```diff
@@ -23,194 +23,259 @@
 
-    # Send a message to telegram
-    def send_message(self, chat_id, text):
+    def send_sticker(self, token, chat_id, sticker_url,
+                     max_attempts=3, notify=False):
         args = {
-            'chat_id': chat_id,
-            'text': text,
-            'disable_web_page_preview': 'False',
-            'disable_notification': 'False',
-            'parse_mode': 'HTML'
+            'url': "https://api.telegram.org/bot{}/sendSticker".format(token),
+            'payload': {
+                'chat_id': chat_id,
+                'sticker': sticker_url,
+                'disable_notification': not notify
+            }
         }
-        try_sending(log, self.connect,
-                    "Telegram", self.__client.sendMessage, args)
+        try_sending(
+            log, self.connect, "Telegram (STKR)", self.send_webhook, args,
+            max_attempts)
 
-    # Send a sticker to telegram
-    def send_sticker(self, chat_id, sticker_id):
+    def send_message(self, token, chat_id, message,
+                     max_attempts=3, notify=True):
         args = {
-            'chat_id': chat_id,
-            'sticker': unicode(sticker_id),
-            'disable_notification': 'True'
+            'url': "https://api.telegram.org/bot{}/sendMessage".format(token),
+            'payload': {
+                'chat_id': chat_id,
+                'text': message,
+                'parse_mode': 'Markdown',
+                'disable_web_page_preview': False,
+                'disable_notification': not notify
+            }
         }
-        try_sending(log, self.connect, 'Telegram (sticker)',
-                    self.__client.sendSticker, args)
+        try_sending(
+            log, self.connect, "Telegram (MSG)", self.send_webhook, args,
+            max_attempts)
 
-    # Send a venue message to telegram
-    def send_venue(self, alert, info):
+    def send_location(self, token, chat_id, lat, lng,
+                      max_attempts=3, notify=False):
         args = {
-            'chat_id': alert['chat_id'],
-            'latitude': info['lat'],
-            'longitude': info['lng'],
-            'title': replace(alert['title'], info),
-            'address': replace(alert['body'], info),
-            'disable_notification': 'False'
+            'url': "https://api.telegram.org/bot{}/sendLocation".format(token),
+            'payload': {
+                'chat_id': chat_id,
+                'latitude': lat,
+                'longitude': lng,
+                'disable_notification': not notify
+            }
         }
-        try_sending(log, self.connect, "Telegram (venue)",
-                    self.__client.sendVenue, args)
+        try_sending(
+            log, self.connect, "Telegram (LOC)", self.send_webhook, args,
+            max_attempts)
 
-    # Send a location message to telegram
-    def send_location(self, alert, info):
+    def send_venue(self, token, chat_id, lat, lng, max_attempts):
```

Comment:
The new send_venue lacks the 'message' text. You're only passing/using chat_id and the location. This will send a venue without any text.

Reply:
Yeah that is also something I need to add. My intent is to split it on the first new line.